executor/aggfuncs/func_group_concat.go

```diff
@@ -82,29 +82,28 @@ func (e *groupConcat) ResetPartialResult(pr PartialResult) {
 
 func (e *groupConcat) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) (err error) {
 	p := (*partialResult4GroupConcat)(pr)
-	v, isNull := "", false
+	v, isNull, preLen := "", false, 0
 	for _, row := range rowsInGroup {
-		if p.buffer != nil {
+		if p.buffer != nil && p.buffer.Len() != 0 {
+			preLen = p.buffer.Len()
 			p.buffer.WriteString(e.sep)
 		}
-		isAllNull := true
 		for _, arg := range e.args {
 			v, isNull, err = arg.EvalString(sctx, row)
 			if err != nil {
 				return errors.Trace(err)
 			}
 			if isNull {
-				continue
+				break
```

Comment:
Why change here?

Reply:
A bug-fix.
Test case is here. https://github.com/pingcap/tidb/pull/7268/files#diff-ab83e4a8f21f295ae805f30d7e889703R332